tests/mlmodel_openai/test_chat_completion.py

```diff
@@ -13,19 +13,184 @@
+        {'type': 'LlmChatCompletionMessage'},
+        {
+            'id': "chatcmpl-87sb95K4EF2nuJRcTs43Tm9ntTemv-0",
+            'appName': 'Python Agent Test (mlmodel_openai)',
+            'request_id': "49dbbffbd3c3f4612aa48def69059ccd",
+            'span_id': "span-id",
+            'trace_id': "trace-id",
+            'transaction_id': None,
+            'content': 'You are a scientist.',
+            'role': 'system',
+            'completion_id': None,
+            'sequence': 0,
+            'response.model': 'gpt-3.5-turbo-0613',
+            'vendor': 'openAI',
+            'ingest_source': 'Python'
+        },
+    ),
+    (
+        {'type': 'LlmChatCompletionMessage'},
+        {
+            'id': "chatcmpl-87sb95K4EF2nuJRcTs43Tm9ntTemv-1",
+            'appName': 'Python Agent Test (mlmodel_openai)',
+            'request_id': "49dbbffbd3c3f4612aa48def69059ccd",
+            'span_id': "span-id",
+            'trace_id': "trace-id",
+            'transaction_id': None,
+            'content': 'What is 212 degrees Fahrenheit converted to Celsius?',
+            'role': 'user',
+            'completion_id': None,
+            'sequence': 1,
+            'response.model': 'gpt-3.5-turbo-0613',
+            'vendor': 'openAI',
+            'ingest_source': 'Python'
+        },
+    ),
+    (
+        {'type': 'LlmChatCompletionMessage'},
+        {
+            'id': "chatcmpl-87sb95K4EF2nuJRcTs43Tm9ntTemv-2",
+            'appName': 'Python Agent Test (mlmodel_openai)',
+            'request_id': "49dbbffbd3c3f4612aa48def69059ccd",
+            'span_id': "span-id",
+            'trace_id': "trace-id",
+            'transaction_id': None,
+            'content': '212 degrees Fahrenheit is equal to 100 degrees Celsius.',
+            'role': 'assistant',
+            'completion_id': None,
+            'sequence': 2,
+            'response.model': 'gpt-3.5-turbo-0613',
+            'vendor': 'openAI',
+            'ingest_source': 'Python'
+        }
+    ),
+]
+
+
+@reset_core_stats_engine()
+@validate_ml_events(sync_chat_completion_recorded_events)
+# One summary event, one system message, one user message, and one response message from the assistant
+@validate_ml_event_count(count=4)
+@background_task()
+def test_openai_chat_completion_sync_in_txn():
+    set_trace_info()
+    add_custom_attribute("conversation_id", "my-awesome-id")
+    openai.ChatCompletion.create(
+        model="gpt-3.5-turbo",
+        messages=_test_openai_chat_completion_sync_messages,
+        temperature=0.7,
+        max_tokens=100
+    )
+
+
+@reset_core_stats_engine()
+@validate_ml_event_count(count=0)
+def test_openai_chat_completion_sync_outside_txn():
+    set_trace_info()
+    add_custom_attribute("conversation_id", "my-awesome-id")
+    openai.ChatCompletion.create(
+        model="gpt-3.5-turbo",
+        messages=_test_openai_chat_completion_sync_messages,
+        temperature=0.7,
+        max_tokens=100
+    )
+
+
+disabled_ml_settings = {
```

Comment:
Is there something in the CDD about putting this behind an on-off switch like machine_learning.enabled? If there's not I'd say let's just leave that out of the implementation and tests for now so this would just be:
disabled_ml_event_settings = {  # Should not send events
    "ml_insights_events.enabled": False
}

If there is, then we should only record an event if `ml_insights_events.enabled=True`. And we should test these settings independently so have a test for:
```
disabled_ml_settings = {  # Should not send events
    "machine_learning.enabled": True,
    "ml_insights_events.enabled": False
}
disabled_ml_settings = {  # Should not send events
    "machine_learning.enabled": False,
    "ml_insights_events.enabled": True
}
disabled_ml_settings = {  # Should send events.
    "machine_learning.enabled": True,
    "ml_insights_events.enabled": True
}
```

Reply:
We don't have any mention of gating this behind `machine_learning.enabled` (at least currently). I will remove that from the tests... we can add in the setting combo tests if we decide to bring this in later.